src/Passes/README.md

```diff
@@ -0,0 +1,327 @@
+
+# Getting started
+
+## Dependencies
+
+This library is written in C++ and depends on:
+
+- LLVM
+
+Additional development dependencies include:
+
+- CMake
+- clang-format
+- clang-tidy
+
+## Building the passes
+
+To build the passes, create a new build directory and switch to that directory:
+
+```sh
+mkdir Debug
+cd Debug/
+```
+
+To build the library, first configure CMake from the build directory
+
+```sh
+cmake ..
+```
+
+and then make your target
+
+```sh
+make [target]
+```
+
+Valid targets are the name of the folders in `libs/` found in the passes root.
+
+## Running a pass
+
+You can run a pass using [opt](https://llvm.org/docs/CommandGuide/opt.html) as follows:
+
+```sh
+cd examples/ClassicalIrCommandline
+make emit-llvm-bc
+opt -load-pass-plugin ../../{Debug,Release}/libOpsCounter.{dylib,so} --passes="print<operation-counter>" -disable-output classical-program.bc
+```
+
+For a detailed tutorial, see examples.
+
+## Creating a new pass
+
+To make it easy to create a new pass, we have created a few templates to get you started quickly:
+
+```sh
+% ./manage create-pass HelloWorld
+Available templates:
+
+1. Function Pass
+
+Select a template:1
+```
+
+At the moment you only have one choice which is a function pass. Over time we will add additional templates. Once you have instantiated your template, you are ready to build it:
+
+```sh
+% mkdir Debug
+% cd Debug
+% cmake ..
+-- The C compiler identification is AppleClang 12.0.5.12050022
+-- The CXX compiler identification is AppleClang 12.0.5.12050022
+(...)
+-- Configuring done
+-- Generating done
+-- Build files have been written to: ./qsharp-compiler/src/Passes/Debug
+
+% make
+
+[ 25%] Building CXX object libs/CMakeFiles/OpsCounter.dir/OpsCounter/OpsCounter.cpp.o
+[ 50%] Linking CXX shared library libOpsCounter.dylib
+[ 50%] Built target OpsCounter
+[ 75%] Building CXX object libs/CMakeFiles/HelloWorld.dir/HelloWorld/HelloWorld.cpp.o
+[100%] Linking CXX shared library libHelloWorld.dylib
+[100%] Built target HelloWorld
+```
+
+Your new pass is ready to be implemented. Open `libs/HelloWorld/HelloWorld.cpp` to implement the details of the pass. At the moment, the
+template will not do much except for print the function names of your code. To test your new pass go to the directory `examples/ClassicalIrCommandline`,
+build an IR and run the pass:
+
+```sh
+% cd ../examples/ClassicalIrCommandline
+% make
+% opt -load-pass-plugin ../../Debug/libs/libHelloWorld.{dylib,so} --passes="hello-world" -disable-output classical-program.ll
+```
+
+If everything worked, you should see output like this:
+
+```sh
+Implement your pass here: foo
+Implement your pass here: bar
+Implement your pass here: main
+```
+
+## CI
+
+Before making a pull request with changes to this library, please ensure that style checks passes, that the code compiles,
+unit test passes and that there are no erros found by the static analyser.
+
+To setup the CI environment, run following commands
```

Comment:
Here too the instructions are Linux specific. If we do not support dev/test on Windows right now, we should explicitly call that out and direct readers to set up WSL or containers to use a Linux development environment (ideally pointing to the VSCode devcontainer at the top of the repo and then updating that to have the necessary prerequisites).

Reply:
Will be addressed in a later PR that is CI focused. No need to update that here now!